cloudferrylib/os/network/neutron.py

```diff
@@ -937,12 +940,13 @@ def upload_lb_pools(self, pools, subnets):
                             'name': pool['name'],
                             'description': pool['description'],
                             'tenant_id': tenant_id,
-                            'provider': pool['provider'],
                             'subnet_id': snet_id,
                             'protocol': pool['protocol'],
                             'lb_method': pool['lb_method']
                         }
                 }
+                if pool.get('provider'):
```

Comment:
Let's always keep `provider` key in `pool` in order to avoid `KeyError` during `deploy()`:
```
# no conditions here
pool_info['pool']['provider'] = pool.get('provider')
```